search/query/boolean.go

```diff
@@ -33,19 +34,17 @@ type booleanQuery struct {
 // Queries.
 // Result documents that ALSO satisfy any of the should
 // Queries will score higher.
-func NewBooleanQuery(must []Query, should []Query, mustNot []Query) *booleanQuery {
+func NewBooleanQuery(must []Query, should []Query, mustNot []Query) *BooleanQuery {
 	return NewBooleanQueryMinShould(must, should, mustNot, 0.0)
 }
 
 // NewBooleanQueryMinShould is the same as
 // NewBooleanQuery, only it offers control of the
 // minimum number of should queries that must be
 // satisfied.
-func NewBooleanQueryMinShould(must []Query, should []Query, mustNot []Query, minShould float64) *booleanQuery {
+func NewBooleanQueryMinShould(must []Query, should []Query, mustNot []Query, minShould float64) *BooleanQuery {
```

Comment:
Yeah, I'd be in favor of that, there is a SetMinShould() method on the BooleanQuery, so it can always be done in 2 steps.

Similarly there is NewMatchQueryOperator() which does not have a setter method to toggle, but I think if it did, that constructor could go away too.  I'll see about packaging this up into another commit.